File: Java/NetBeansProjects/Java Basics/anil kumar/binarysearch.java
import java.util.Scanner;
class binarysearch
{
    public static void main( String a[])
    {
        int i=0;
        System.out.println("enter the size ");
        Scanner sc =new Scanner(System.in);
         int n = sc.nextInt( );
          int b[]=new int[n]; 
        System.out.println("enter the elements ");
        for(int k=0;k<n;k++) 
        {
            b[k]=sc.nextInt( );
        }
        
        for(int j=0;j<n-1;j++)
        {
           if(b[j]>b[j+1])
           {
               int temp= b[j];
               b[j]= b[j+1];
               b[j+1]=temp;
           } 
        }
        int last = n-1;
        int mid = (i+last)/2;
        System.out.println("enter the element you want to find");
         int x = sc.nextInt( );
      
        while(i <= last)
        {
            if(b[mid] < x)
            {
                i = mid+1;
            }
            else if(b[mid] == x)
            {
                System.out.println("element found at pos " + (mid+1));
                break;
            }
            else
           {
               last = mid-1;
           }
            mid=(i+last)/2;
        }
        if(i > last)
        {
            {
                System.out.println("element not found ");
            }
        }
    }
}

File: DataStructure/ProblemSET/binarySearch.cpp
//binary Search using pair for unsorted array
#include<iostream>
#include<algorithm>
using namespace std;

void binarySearch(vector<pair<int,int>> v){
    int x,first=0,last=v.size()-1,count=0,mid;
    cout<<"Enter the element you want to find: ";
    cin>>x;
    while(first<=last){
        mid=(first+last)/2;
        if(v[mid].first>x){
            last=mid-1;
        }
        if(v[mid].first<x){
            first=mid+1;
        }
        if(v[mid].first==x){
            count++;
            break;
        }
    }
    if(count){
        cout<<"The element "<<v[mid].first<<" is found at position: "<<v[mid].second+1;
    }
    else{
        cout<<"The element is not found"<<endl;
    }
}
bool mycompare(pair<int,int> p1,pair<int,int> p2){
	return p1.first<p2.first;
}
int main(){
    int *a,n;
    cout<<"Enter the size of a array: ";
    cin>>n;
    a=new int[n];
//    cout<<"Enter the Sorted elements: ";
    for(int i=0;i<n;i++){
        cin>>a[i];
    }
    vector<pair<int,int>> v;
    for(int i=0;i<n;i++){
    		v.push_back(make_pair(a[i],i));
    }
    sort(v.begin(),v.end(),mycompare);
    binarySearch(v);
}
